driver/src/main/java/eu/cloudnetservice/cloudnet/driver/permission/CachedPermissionManagement.java

```diff
@@ -21,29 +21,121 @@
 
+  /**
+   * Gets a map with all cached permission users mapped to their uuid.
+   *
+   * @return all cached permission users.
+   */
   @NonNull Map<UUID, PermissionUser> cachedPermissionUsers();
 
+  /**
+   * Gets a map with all cached permission groups mapped to their name.
+   *
+   * @return all cached permission groups.
+   */
   @NonNull Map<String, PermissionGroup> cachedPermissionGroups();
 
+  /**
+   * Searches the permission user cache for a permission user with the given unique id.
+   *
+   * @param uniqueId the unique id associated with the permission user.
+   * @return the cached permission user for the given unique id, null if no user was found in the cache.
+   * @throws NullPointerException if the given unique id is null.
+   */
   @Nullable PermissionUser cachedUser(@NonNull UUID uniqueId);
 
+  /**
+   * Searches the permission group cache for a permission group with the given name.
+   *
+   * @param name the name associated with the permission group.
+   * @return the cached permission group for the given name, null if no group was found in the cache.
+   * @throws NullPointerException if the given name is null.
+   */
   @Nullable PermissionGroup cachedGroup(@NonNull String name);
 
+  /**
+   * Acquires a lock for the permission user. The lock ensures that the user stays in cache.
+   * <p>
+   * Each method call results in incrementing the lock count of the user. The user stays in cache until the lock count
+   * is down to 0.
+   *
+   * @param user the user to acquire the lock for.
+   * @throws NullPointerException if the given user is null.
+   */
   void acquireLock(@NonNull PermissionUser user);
 
+  /**
+   * Acquires a lock for the permission group. The lock ensures that the group stays in cache.
+   * <p>
+   * Each method call results in incrementing the lock count of the group. The group stays in cache until the lock count
+   * is down to 0.
+   *
+   * @param group the group to acquire the lock for.
+   * @throws NullPointerException if the given group is null.
+   */
   void acquireLock(@NonNull PermissionGroup group);
 
+  /**
+   * Checks if the given user has any locks locking the user in the cache.
+   *
+   * @param user the user to check for locks.
+   * @return true if there are any locks for the given user, false otherwise.
+   * @throws NullPointerException if the given user is null.
+   */
   boolean locked(@NonNull PermissionUser user);
 
+  /**
+   * Checks if the given group has any locks locking the group in the cache.
+   *
+   * @param group the group to check for locks.
+   * @return true if there are any locks for the given group, false otherwise.
+   * @throws NullPointerException if the given group is null.
+   */
   boolean locked(@NonNull PermissionGroup group);
 
+  /**
+   * Removes exactly one lock of the user. If the user does not have any locks no changes are made. If there are no
+   * remaining locks the user is removed from the cache after a maximum of 5 minutes.
```

Comment:
I don't think that we should specifiy the minute count here. I would suggest using the same generified sentence as for the two methods above 🤔